neuraxle/metaopt/random.py

```diff
@@ -25,27 +25,264 @@
+        self.scores_train = self.scoring_function(expected_outputs, data_inputs)
+        self.scores_train_mean = np.mean(self.scores_train)
+        self.scores_train_std = np.std(self.scores_train)
+
+    def split_data_container(self, data_container) -> Tuple[DataContainer, DataContainer]:
+        """
+        Split data container into a training set, and a validation set.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :return: train_data_container, validation_data_container
+        """
+
+        train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs = \
+            self.split(data_container.data_inputs, data_container.expected_outputs)
+
+        train_ids = self.train_split(data_container.current_ids)
+        train_data_container = DataContainer(train_ids, train_data_inputs, train_expected_outputs)
+
+        validation_ids = self.validation_split(data_container.current_ids)
+        validation_data_container = DataContainer(validation_ids, validation_data_inputs, validation_expected_outputs)
+
+        return train_data_container, validation_data_container
+
+    def split(self, data_inputs, expected_outputs=None) -> Tuple[List, List, List, List]:
+        """
+        Split data inputs, and expected outputs into a training set, and a validation set.
+
+        :param data_inputs: data inputs to split
+        :param expected_outputs: expected outputs to split
+        :return: train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+        """
+        validation_data_inputs = self.validation_split(data_inputs)
+        validation_expected_outputs = None
+        if expected_outputs is not None:
+            validation_expected_outputs = self.validation_split(expected_outputs)
+
+        train_data_inputs = self.train_split(data_inputs)
+        train_expected_outputs = None
+        if expected_outputs is not None:
+            train_expected_outputs = self.train_split(expected_outputs)
+
+        return train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+
+    def train_split(self, data_inputs) -> List:
+        """
+        Split training set.
+
+        :param data_inputs: data inputs to split
+        :return: train_data_inputs
+        """
+        index_split = math.floor(len(data_inputs) * (1 - self.test_size))
+        return data_inputs[0:index_split]
+
+    def validation_split(self, data_inputs) -> List:
+        """
+        Split validation set.
+
+        :param data_inputs: data inputs to split
+        :return: validation_data_inputs
+        """
+        index_split = math.floor(len(data_inputs) * (1 - self.test_size))
+        validation_data_inputs = data_inputs[index_split:]
+        return validation_data_inputs
+
+
+class BaseCrossValidationWrapper(BaseValidation, ABC):
+    # TODO: assert that set_step was called.
```

Comment:
Maybe this assert could be done in the fit method. Or maybe even worse: in the setup method, to catch errors early on. Raise some kind of errors, maybe not an assertion error.

Reply:
You can resolve this to issue but we need an issue.